pkg/monitor/cluster/clusteroperatorsversions.go

```diff
@@ -0,0 +1,41 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+func (mon *Monitor) emitClusterOperatorsVersions(ctx context.Context) error {
+	cos, err := mon.configcli.ConfigV1().ClusterOperators().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(cos.Items, func(i, j int) bool { return cos.Items[i].Name < cos.Items[j].Name })
+	for _, co := range cos.Items {
+		for _, v := range co.Status.Versions {
+			if v.Name != "operator" {
+				continue
+			}
+
+			if v.Version == version.OpenShiftVersion {
+				continue
+			}
```

Comment:
I don't think so. 
This way if customer configured `4.4`, and all operators somehow got there we will not know as this metric will be valid from this point of view.

I think this indicates bad cluster earlier, where we might have RP on `4.3.12` and operators on a higher version.

What we gonna do with this - this is a different question.

Reply:
+if cluster desired version is still `4.2` we know upgrades is in flux too... so more data points

Reply:
won't this cause a metrics explosion when we upgrade the RPs to a new target OpenShift version?
whereas if we check against the cluster desired version, the explosion is smaller because it only happens as we upgrade clusters.
we will know the clusters which don't match the target RP version via clusterversions.

Reply:
I think this can be done both way. Changed :man_shrugging: